### ISSUE TITLE
vizier query_region returning empty TableList

### ISSUE THREAD

**Issue by JohnPritchard** [astropy/astroquery#1962]
Hi,

Since at least Monday 25/01/2021, query_region requests to Vizier have been returning empty TableLists: e.g.:

> from astroquery.vizier import Vizier
> from astropy import units as u
> from astropy.coordinates.sky_coordinate import SkyCoord
> c=SkyCoord(185.338, -60.4006, unit=('deg','deg'))
> v=Vizier(columns=["**", "+_r"], row_limit=-1)
> q=v.query_region(
>       c,
>       radius=720.0*u.arcsec,
>       catalog=['I/345/gaia2','I/289/out',],
>       cache=False,
>    )
> print(q)

I am getting this with a range of astroquery versions, 0.4 to 0.4.2.dev6449, and in both macOS and Linux environments, so I am guessing Vizier has changed something in their API.

Any ideas how to fix this?

**Comment by keflavich** [comment]
I confirm: this looks like a bug, maybe an upstream issue?  If I search the gaia2 catalog at these coordinates, there are _lots_ of hits, so it's not a genuinely empty catalog.

cc the Vizier maintainers, I need to get their names linked...

**Comment by JohnPritchard** [comment]
I have reached out to the CDS to ask for info, maybe it's simply that they have some down servers (has happened before), or maybe they did something to the "API" on their end. Will keep you posted.

**Comment by JohnPritchard** [comment]
Today it seems to be working again. So I am guessing it was an issue on the CDS end, and my email to them prompted them to take action...
This ticket can probably be closed now.

**Comment by keflavich** [comment]
@gilleslandais could you verify that you guys solved this?  Is it related to #1826?

**Comment by ivv101** [comment]
It still doesn't work: even the example in the documentation --https://astroquery.readthedocs.io/en/latest/vizier/vizier.html#query-with-table -- produces an empty TableList. 
I was wondering if it could be mentioned in the documentation  that querying with Tables is temporarily not working -- would be very helpful.

We need to query several thousands of coordinates and because tables don't work, we query a coordinate at a time in a loop and It takes quite a long time. Would querying with Tables retrieve data faster than synchronous (or asynchronous) one coordinate queries? Thank you.

**Comment by keflavich** [comment]
@ivv101 we don't put temporary issues into the docs, generally - that's what the issue tracker is for.

This might be an upstream issue.  I'll ask CDS for help.

**Comment by keflavich** [comment]
@gilleslandais reported (by email) that this is a potentially-recurring issue having to do with switching between the "main" and "backup" servers.   If we query using TAP instead of ADQL, we may be able to get this error message: `<INFO name="QUERY_STATUS" value="ERROR">Upload aborted: this functionality is disabled in this TAP service!</INFO>` and pass it on to users, but at the moment it seems our current strategy of using the Vizier web adql interface means we do not get clear error messages, so there's nothing we can do about this right now.

**Comment by jwoillez** [comment]
@ivv101 @keflavich I have a fix for this query-with-table issue. It seems that the upstream query-with-list mechanism changed.

**Comment by pacargile** [comment]
@jwoillez Any update on your PR to fix this issue? I really don't want to get black-listed from Vizier because I spammed their servers with thousands of individual queries.

**Comment by jwoillez** [comment]
I just added a note to #2012. I may have forgotten to state that the PR was ready for review. It also look like it is the busy season, away from astroquery. @pacargile, you can always use the branch used for the PR in the meantime. And report if it works for you in #2012 or here. Would be nice if other users test that branch.

**Comment by pacargile** [comment]
@jwoillez Using your fork, I was able to get the example [query-with-table](https://astroquery.readthedocs.io/en/latest/vizier/vizier.html#query-with-table) to work without issue. However, I tried it with my own query to Gaia ER3 with an astropy SkyCoord object. I ran into a VOTable parsing error:

```python
import astropy.units as u
from astropy.coordinates import SkyCoord
from astroquery.vizier import Vizier

RA  = [ 81.05389846033063,80.80170458981672]
DEC = [-2.2564481640418355,-2.8452267772213875]

coord = SkyCoord(ra=RA, dec=DEC, unit=u.deg, frame="icrs")

r = Vizier(columns=["**", "+_r"]).query_region(coord,radius="10s",catalog='I/350/gaiaedr3')

```

The result is:

```command
/Users/pcargile/Library/Python/3.6/lib/python/site-packages/astroquery-0.4.2.dev6640-py3.6.egg/astroquery/vizier/core.py:734: UserWarning: VOTABLE parsing raised exception: None:413:8: ValueError: invalid literal for int() with base 10: '0825-01478473' (in row 100, col '_q')
  warnings.warn("VOTABLE parsing raised exception: {0}".format(ex))
Traceback (most recent call last):
  File "/Users/pcargile/Library/Python/3.6/lib/python/site-packages/astroquery-0.4.2.dev6640-py3.6.egg/astroquery/vizier/core.py", line 732, in parse_vizier_votable
    vo_tree = votable.parse(tf, pedantic=False, invalid='exception')
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/astropy/utils/decorators.py", line 535, in wrapper
    return function(*args, **kwargs)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/astropy/io/votable/table.py", line 167, in parse
    config=config, pos=(1, 1)).parse(iterator, config)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/astropy/io/votable/tree.py", line 3575, in parse
    iterator, tag, data, config, pos)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/astropy/io/votable/tree.py", line 3485, in _add_resource
    resource.parse(self, iterator, config)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/astropy/io/votable/tree.py", line 3287, in parse
    iterator, tag, data, config, pos)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/astropy/io/votable/tree.py", line 3232, in _add_table
    table.parse(iterator, config)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/astropy/io/votable/tree.py", line 2507, in parse
    iterator, colnumbers, fields, config)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/astropy/io/votable/tree.py", line 2622, in _parse_tabledata
    vo_reraise(e, config, pos)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/astropy/io/votable/exceptions.py", line 130, in vo_reraise
    raise exc
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/astropy/io/votable/tree.py", line 2619, in _parse_tabledata
    fields[i].ID))
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/astropy/io/votable/exceptions.py", line 130, in vo_reraise
    raise exc
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/astropy/io/votable/tree.py", line 2613, in _parse_tabledata
    data, config, pos)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/astropy/io/votable/converters.py", line 840, in parse
    value = int(value, 10)
ValueError: None:413:8: ValueError: invalid literal for int() with base 10: '0825-01478473' (in row 100, col '_q')

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/pcargile/Library/Python/3.6/lib/python/site-packages/astroquery-0.4.2.dev6640-py3.6.egg/astroquery/vizier/core.py", line 666, in _parse_result
    get_catalog_names=get_catalog_names)
  File "/Users/pcargile/Library/Python/3.6/lib/python/site-packages/astroquery-0.4.2.dev6640-py3.6.egg/astroquery/vizier/core.py", line 735, in parse_vizier_votable
    vo_tree = votable.parse(tf, pedantic=False, invalid='mask')
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/astropy/utils/decorators.py", line 535, in wrapper
    return function(*args, **kwargs)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/astropy/io/votable/table.py", line 167, in parse
    config=config, pos=(1, 1)).parse(iterator, config)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/astropy/io/votable/tree.py", line 3575, in parse
    iterator, tag, data, config, pos)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/astropy/io/votable/tree.py", line 3485, in _add_resource
    resource.parse(self, iterator, config)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/astropy/io/votable/tree.py", line 3287, in parse
    iterator, tag, data, config, pos)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/astropy/io/votable/tree.py", line 3232, in _add_table
    table.parse(iterator, config)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/astropy/io/votable/tree.py", line 2507, in parse
    iterator, colnumbers, fields, config)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/astropy/io/votable/tree.py", line 2634, in _parse_tabledata
    vo_raise(E21, (i, len(fields)), config, pos)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/astropy/io/votable/exceptions.py", line 112, in vo_raise
    raise exception_class(args, config, pos)
astropy.io.votable.exceptions.E21: None:413:112: E21: Data has fewer columns (8) than are defined in the header (68)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "tmp.py", line 20, in <module>
    r = Vizier(columns=["**", "+_r"]).query_region(coord,radius="10s",catalog='II/349/ps1')
  File "/Users/pcargile/Library/Python/3.6/lib/python/site-packages/astroquery-0.4.2.dev6640-py3.6.egg/astroquery/utils/class_or_instance.py", line 25, in f
    return self.fn(obj, *args, **kwds)
  File "/Users/pcargile/Library/Python/3.6/lib/python/site-packages/astroquery-0.4.2.dev6640-py3.6.egg/astroquery/utils/process_asyncs.py", line 29, in newmethod
    result = self._parse_result(response, verbose=verbose)
  File "/Users/pcargile/Library/Python/3.6/lib/python/site-packages/astroquery-0.4.2.dev6640-py3.6.egg/astroquery/vizier/core.py", line 676, in _parse_result
    str(self.table_parse_error))
astroquery.exceptions.TableParseError: Failed to parse VIZIER result! The raw response can be found in self.response, and the error in self.table_parse_error. The attempted parsed result is in self.parsed_result.
 Exception: None:413:112: E21: Data has fewer columns (8) than are defined in the header (68)
```

My astropy version is 4.1.

Thanks for your work on this issue!

**Comment by pacargile** [comment]
@jwoillez I've tracked down the issue having to do with giving the Vizier object (e.g., Vizier(columns=["**", "+_r"]) ). My example above works when the query is simplified to:

```python
import astropy.units as u
from astropy.coordinates import SkyCoord
from astroquery.vizier import Vizier

RA  = [ 81.05389846033063,80.80170458981672]
DEC = [-2.2564481640418355,-2.8452267772213875]

coord = SkyCoord(ra=RA, dec=DEC, unit=u.deg, frame="icrs")

r = Vizier.query_region(coord,radius="10s",catalog='I/350/gaiaedr3')
```

**Comment by jwoillez** [comment]
@pacargile Thanks for the bug report. Good thing you could test. I try to push a fix over the week-end.

**Comment by jwoillez** [comment]
@pacargile A fix for your issue was added to PR #2012. Want to try again?
